### PR TITLE
Pv/retain bbox intersects parameters

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** 

- https://github.com/stac-utils/stac-server/issues/99
- https://github.com/stac-utils/stac-server/issues/176
- https://github.com/stac-utils/stac-server/issues/164
- https://github.com/stac-utils/stac-server/issues/165

**Proposed Changes:**

1. fix next/prev pagination links to have bbox parameter in them if the request had it (previously, the bbox was converted to an intersects in the links)
2. fix bbox validation to return 400 error for invalid bbox

**PR Checklist:**

- [X] I have added my changes to the [CHANGELOG](https://github.com/stac-utils/stac-server/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: src/lambdas/api/app.js [hunk 1]
```diff
@@ -179,21 +191,22 @@ app.put('/collections/:collectionId/items/:itemId', async (req, res, next) => {
...
-        await api.updateItem(req.body, es)
-        res.sendStatus(204)
-      } catch (error) {
-        if (error instanceof Error
-              && error.name === 'ResponseError'
-              && error.message.includes('version_conflict_engine_exception')) {
-          res.sendStatus(409)
-        } else {
-          next(error)
+      const itemRes = await api.getItem(collectionId, itemId, es, req.endpoint)
```
Comment (philvarner) [inline src/lambdas/api/app.js:194]: validate the item exists for a put rather than just the collection (should have been pulled into another PR, but here we are)

File: tests/system/test-api-item-post.js [hunk 2]
```diff
@@ -67,17 +67,6 @@ test('POST /collections/:collectionId/items', async (t) => {
   t.assert(response.headers['location'].endsWith(`/collections/${collectionId}/items/${itemId}`))
   t.is(response.body, 'Created')
 
-  // now test mismatched collection ids, while we wait a second for our item propagate
```
Comment (philvarner) [inline tests/system/test-api-item-post.js:70]: this had been pulled into another test, but not deleted here.

File: tests/system/test-api-item-put.js [hunk 3]
```diff
@@ -77,12 +77,12 @@ test('PUT /collections/:collectionId/items/:itemId', async (t) => {
 test('PUT /collections/:collectionId/items/:itemId for a non-existent collection or id returns 404"', async (t) => {
   const { collectionId } = t.context
 
-  t.is((await t.context.api.client.patch(
+  t.is((await t.context.api.client.put(
```
Comment (philvarner) [inline tests/system/test-api-item-put.js:80]: missed updating these tests when copied the patch tests to put tests.

**Review by matthewhanson — APPROVED** [review]
